test/celeritas/global/StepperTestBase.cc

```diff
@@ -0,0 +1,216 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/global/StepperTestBase.cc
+//---------------------------------------------------------------------------//
+#include "StepperTestBase.hh"
+
+#include <algorithm>
+#include <cstring>
+#include <numeric>
+#include <gtest/gtest.h>
+
+#include "corecel/io/Repr.hh"
+#include "celeritas/global/ActionManager.hh"
+#include "celeritas/global/Stepper.hh"
+
+#include "celeritas_cmake_strings.h"
+
+using namespace celeritas;
+using std::cout;
+
+namespace celeritas_test
+{
+namespace
+{
+//---------------------------------------------------------------------------//
+//! Test for equality of two C strings
+bool cstring_equal(const char* lhs, const char* rhs)
+{
+    return std::strcmp(lhs, rhs) == 0;
+}
+//---------------------------------------------------------------------------//
+} // namespace
+
+//---------------------------------------------------------------------------//
+//! Whether Geant4 dependencies match those on the CI build
+bool StepperTestBase::is_ci_build()
+{
+    return cstring_equal(celeritas_rng, "XORWOW")
+           && cstring_equal(celeritas_clhep_version, "2.4.4.0")
```

Comment:
Why do we use a specific (system) version of clhep instead of one that comes with Geant4 for celeritas testings?

Reply:
The CI systems (and the environments I've installed on wildstyle and my work computer) use spack which can have different CLHEP and Geant4 versions. We can't guarantee that the versions are locked so I just added the clhep version to be safer. We'd ideally be testing the g4emlow (and other geant4 dataset) versions too but there seems to be no easy way to access those.